rpc/src/main/scala/org/bitcoins/rpc/client/RpcClient.scala

```diff
@@ -0,0 +1,1114 @@
+package org.bitcoins.rpc.client
+
+import java.net.URI
+
+import akka.http.javadsl.model.headers.HttpCredentials
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.stream.ActorMaterializer
+import akka.util.ByteString
+import org.bitcoins.core.crypto.{DoubleSha256Digest, ECPrivateKey, ECPublicKey}
+import org.bitcoins.core.currency.{Bitcoins, Satoshis}
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.{BitcoinAddress, P2PKHAddress}
+import org.bitcoins.core.protocol.blockchain.{Block, BlockHeader, MerkleBlock}
+import org.bitcoins.core.protocol.script.ScriptPubKey
+import org.bitcoins.core.protocol.transaction.{
+  Transaction,
+  TransactionInput,
+  TransactionOutPoint
+}
+import org.bitcoins.core.util.{BitcoinSLogger, BitcoinSUtil}
+import org.bitcoins.rpc.client.RpcOpts.AddressType
+import org.bitcoins.rpc.client.RpcOpts.AddressType.AddressType
+import org.bitcoins.rpc.config.DaemonInstance
+import play.api.libs.json._
+import org.bitcoins.rpc.jsonmodels._
+import org.bitcoins.rpc.serializers.JsonSerializers._
+import org.bitcoins.rpc.serializers.JsonWriters.mapWrites
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.sys.process._
+
+class RpcClient(instance: DaemonInstance)(
+    implicit
+    m: ActorMaterializer,
+    ec: ExecutionContext) {
```

Comment:
You do not need to pass in a `ec`. You can just create `implicit val ec = m.system.dispatcher`

Reply:
Should it be `m.system.dispatcher` or `m.executionContext`?